ZenPacks/zenoss/ZenPackLib/lib/base/Component.py

```diff
@@ -6,9 +6,44 @@
 # License.zenoss under the directory where your Zenoss product is installed.
 #
 ##############################################################################
-from Products.ZenModel.DeviceComponent import DeviceComponent
-from Products.ZenModel.ManagedEntity import ManagedEntity
+from Products.ZenModel.CPU import CPU as BaseCPU
+from Products.ZenModel.DeviceComponent import DeviceComponent as BaseDeviceComponent
+from Products.ZenModel.ExpansionCard import ExpansionCard as BaseExpansionCard
+from Products.ZenModel.Fan import Fan as BaseFan
+from Products.ZenModel.FileSystem import FileSystem as BaseFileSystem
+from Products.ZenModel.HardDisk import HardDisk as BaseHardDisk
+from Products.ZenModel.HWComponent import HWComponent as BaseHWComponent
+from Products.ZenModel.IpInterface import IpInterface as BaseIpInterface
+from Products.ZenModel.IpRouteEntry import IpRouteEntry as BaseIpRouteEntry
+from Products.ZenModel.IpService import IpService as BaseIpService
+from Products.ZenModel.ManagedEntity import ManagedEntity as BaseManagedEntity
+from Products.ZenModel.OSComponent import OSComponent as BaseOSComponent
+from Products.ZenModel.OSProcess import OSProcess as BaseOSProcess
+from Products.ZenModel.PowerSupply import PowerSupply as BasePowerSupply
+from Products.ZenModel.Service import Service as BaseService
+from Products.ZenModel.TemperatureSensor import TemperatureSensor as BaseTemperatureSensor
+from Products.ZenModel.WinService import WinService as BaseWinService
+
 from ..factory.ComponentTypeFactory import ComponentTypeFactory
 
 
-Component = ComponentTypeFactory('Component', (DeviceComponent, ManagedEntity))
+Component = ComponentTypeFactory('Component', (BaseDeviceComponent, BaseManagedEntity))
+HWComponent = ComponentTypeFactory('HWComponent', (BaseHWComponent,))
+# Backwards-compatibility aliases.
+HardwareComponent = HWComponent
+CPU = ComponentTypeFactory('CPU', (BaseCPU,))
```

Comment:
The "Backwards-compatibility aliases" comment above makes it appear that CPU and the following lines are also backwards-compatibility aliases. Perhaps it would be better to move these lines to the very bottom of the proxy class section to avoid that confusion.

```
# Backwards-compatibility aliases.
HardwareComponent = HWComponent
```